### PR TITLE
Android sdk

### PR DESCRIPTION
**Pull request by albho**

### CODE REVIEW

File: binding/android/Falcon/falcon/src/main/java/ai/picovoice/FalconSegments.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * Getter for diarized segments and their associated metadata.
+     *
+     * @return Diarized segments and their associated metadata.
+     */
+    public Segment[] getSegmentArray() {
+        return segmentArray;
+    }
```
Comment (ksyeo1010) [inline binding/android/Falcon/falcon/src/main/java/ai/picovoice/FalconSegments.java:38]: if this part is just handling the array, we dont really need to create a class for it. Can just put the class below call it `FalconSegment`. Other parts of the code is fine using it as `FalconSegment[]` for the array.

File: demo/android/FalconDemo/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Running the Instrumented Unit Tests
+
+Ensure you have an Android device connected or simulator running. Then run the following from the terminal:
+
+```console
+cd demo/android/FalconDemo
+./gradlew connectedAndroidTest -PpvTestingAccessKey="YOUR_ACCESS_KEY_HERE"
+```
+
+The test results are stored in `falcon-demo-app/build/reports`.
```
Comment (ksyeo1010) [inline demo/android/FalconDemo/README.md:33]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -217,6 +225,43 @@ console.log(segments);
...
+
+To include the Falcon package in your Android project, ensure you have included `mavenCentral()` in your top-level `build.gradle` file and then add the following to your app's `build.gradle`:
+
+```groovy
+dependencies {
+    implementation 'ai.picovoice:falcon-android:${LATEST_VERSION}'
+}
+```
+
+Create an instance of the engine and generate speech:
```
Comment (ksyeo1010) [inline README.md:238]:
```suggestion
Create an instance of the engine and perform speaker diarization on an audio file:
```

File: binding/android/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Falcon requires a valid Picovoice `AccessKey` at initialization. `AccessKey` acts as your credentials when using Falcon SDKs.
+You can get your `AccessKey` for free. Make sure to keep your `AccessKey` secret.
+Signup or login to [Picovoice Console](https://console.picovoice.ai/) to get your `AccessKey`.
+
+## Permissions
+
+To enable AccessKey validation and recording with your Android device's microphone, you must add the following line to your `AndroidManifest.xml` file:
+```xml
+<uses-permission android:name="android.permission.RECORD_AUDIO" />
+<uses-permission android:name="android.permission.INTERNET" />
```
Comment (ksyeo1010) [inline binding/android/README.md:42]: we dont need mic permissions, only internet should be ok

File: .github/workflows/android-perf.yml [hunk 5]
```diff
@@ -33,7 +33,7 @@ jobs:
             procPerformanceThresholdSec: 1.0
           - device: 32bit-android
             initPerformanceThresholdSec: 1.0
-            procPerformanceThresholdSec: 1.0
+            procPerformanceThresholdSec: 7.0
```
Comment (ksyeo1010) [inline .github/workflows/android-perf.yml:36]: how did this from 1 to 7?

Comment (albho) [reply .github/workflows/android-perf.yml:36]: it was never 1, I put that arbitrarily, then ran the AppCenter test to see how long it takes and updated it after

Comment (ksyeo1010) [reply .github/workflows/android-perf.yml:36]: oh i see

**Review by ksyeo1010 — APPROVED** [review]

**Pull request merged**